app/controllers/feedbacks_controller.rb

```diff
@@ -0,0 +1,50 @@
+class FeedbacksController < ApplicationController
+  FEEDBACK_TYPES = %w[freetext satisfaction].freeze
+
+  def create
+    if params[:type].nil? || !FEEDBACK_TYPES.include?(params[:type])
+      raise "Feedback type needs to be specified"
+    end
+
+    @form = params[:type].include?("satisfaction") ? satisfaction_model : freetext_model
+    if @form.valid?
+      if @form.instance_of?(SatisfactionFeedback)
+        SatisfactionFeedback.create!(
+          satisfied: @form.satisfied,
+          level_of_help:,
+          outcome:,
+        )
+      else
+        FreetextFeedback.create!(
+          text: @form.text,
+          page: @form.page,
+          level_of_help:,
+        )
+      end
+    else
+      track_validation_error
```

Comment:
This is kinda an abuse of this method - we don't want to create a "validation_message" analytics event (which is what this method does), because if this line of code gets hit the user _doesn't_ see a validation message. By all means save an analytics event, but I think it should be a custom one.

Reply:
to chat about when pairing